File: SDKs/monarchinitiative.org/1.0.1/python/src/sdk/models/operations/get_goterm_gene_associations.py
import dataclasses
from typing import Optional
from enum import Enum
from ..shared import associationresults as shared_associationresults


@dataclasses.dataclass
class GetGotermGeneAssociationsPathParams:
    id: str = dataclasses.field(metadata={'path_param': { 'field_name': 'id', 'style': 'simple', 'explode': False }})
    
class GetGotermGeneAssociationsRelationshipTypeEnum(str, Enum):
    INVOLVED_IN = "involved_in"
    INVOLVED_IN_REGULATION_OF = "involved_in_regulation_of"
    ACTS_UPSTREAM_OF_OR_WITHIN = "acts_upstream_of_or_within"


@dataclasses.dataclass
class GetGotermGeneAssociationsQueryParams:
    direct: Optional[bool] = dataclasses.field(default=None, metadata={'query_param': { 'field_name': 'direct', 'style': 'form', 'explode': True }})
    evidence: Optional[str] = dataclasses.field(default=None, metadata={'query_param': { 'field_name': 'evidence', 'style': 'form', 'explode': True }})
    exclude_automatic_assertions: Optional[bool] = dataclasses.field(default=None, metadata={'query_param': { 'field_name': 'exclude_automatic_assertions', 'style': 'form', 'explode': True }})
    facet: Optional[bool] = dataclasses.field(default=None, metadata={'query_param': { 'field_name': 'facet', 'style': 'form', 'explode': True }})
    facet_fields: Optional[list[str]] = dataclasses.field(default=None, metadata={'query_param': { 'field_name': 'facet_fields', 'style': 'form', 'explode': True }})
    fetch_objects: Optional[bool] = dataclasses.field(default=None, metadata={'query_param': { 'field_name': 'fetch_objects', 'style': 'form', 'explode': True }})
    relationship_type: Optional[GetGotermGeneAssociationsRelationshipTypeEnum] = dataclasses.field(default=None, metadata={'query_param': { 'field_name': 'relationship_type', 'style': 'form', 'explode': True }})
    rows: Optional[int] = dataclasses.field(default=None, metadata={'query_param': { 'field_name': 'rows', 'style': 'form', 'explode': True }})
    slim: Optional[list[str]] = dataclasses.field(default=None, metadata={'query_param': { 'field_name': 'slim', 'style': 'form', 'explode': True }})
    start: Optional[int] = dataclasses.field(default=None, metadata={'query_param': { 'field_name': 'start', 'style': 'form', 'explode': True }})
    unselect_evidence: Optional[bool] = dataclasses.field(default=None, metadata={'query_param': { 'field_name': 'unselect_evidence', 'style': 'form', 'explode': True }})
    use_compact_associations: Optional[bool] = dataclasses.field(default=None, metadata={'query_param': { 'field_name': 'use_compact_associations', 'style': 'form', 'explode': True }})
    

@dataclasses.dataclass
class GetGotermGeneAssociationsRequest:
    path_params: GetGotermGeneAssociationsPathParams = dataclasses.field()
    query_params: GetGotermGeneAssociationsQueryParams = dataclasses.field()
    

@dataclasses.dataclass
class GetGotermGeneAssociationsResponse:
    content_type: str = dataclasses.field()
    status_code: int = dataclasses.field()
    association_results: Optional[shared_associationresults.AssociationResults] = dataclasses.field(default=None)
    

File: SDKs/monarchinitiative.org/1.0.1/typescript/src/sdk/models/operations/getgotermgeneassociations.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class GetGotermGeneAssociationsPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=id" })
  id: string;
}

export enum GetGotermGeneAssociationsRelationshipTypeEnum {
    InvolvedIn = "involved_in",
    InvolvedInRegulationOf = "involved_in_regulation_of",
    ActsUpstreamOfOrWithin = "acts_upstream_of_or_within"
}


export class GetGotermGeneAssociationsQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=direct" })
  direct?: boolean;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=evidence" })
  evidence?: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=exclude_automatic_assertions" })
  excludeAutomaticAssertions?: boolean;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=facet" })
  facet?: boolean;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=facet_fields" })
  facetFields?: string[];

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=fetch_objects" })
  fetchObjects?: boolean;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=relationship_type" })
  relationshipType?: GetGotermGeneAssociationsRelationshipTypeEnum;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=rows" })
  rows?: number;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=slim" })
  slim?: string[];

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=start" })
  start?: number;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=unselect_evidence" })
  unselectEvidence?: boolean;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=use_compact_associations" })
  useCompactAssociations?: boolean;
}


export class GetGotermGeneAssociationsRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: GetGotermGeneAssociationsPathParams;

  @SpeakeasyMetadata()
  queryParams: GetGotermGeneAssociationsQueryParams;
}


export class GetGotermGeneAssociationsResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  associationResults?: shared.AssociationResults;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;
}
